src/common/NetworkPrivate.cpp

```diff
@@ -54,360 +149,311 @@ QString NetworkData::getHash()
     return this->hash_;
 }
 
-void writeToCache(const std::shared_ptr<NetworkData> &data,
-                  const QByteArray &bytes)
-{
-    if (data->cache_)
-    {
-        QtConcurrent::run([data, bytes] {
-            QFile cachedFile(getPaths()->cacheDirectory() + "/" +
-                             data->getHash());
-
-            if (cachedFile.open(QIODevice::WriteOnly))
-            {
-                cachedFile.write(bytes);
-            }
-        });
-    }
-}
-
 void loadUncached(std::shared_ptr<NetworkData> &&data)
 {
     DebugCount::increase("http request started");
 
     NetworkRequester requester;
-    auto *worker = new NetworkWorker;
+    auto *worker = new NetworkTask(std::move(data));
 
     worker->moveToThread(&NetworkManager::workerThread);
 
-    auto onUrlRequested = [data, worker]() mutable {
-        if (data->hasTimeout_)
-        {
-            data->timer_ = new QTimer();
-            data->timer_->setSingleShot(true);
-            data->timer_->start(data->timeoutMS_);
-        }
+    QObject::connect(&requester, &NetworkRequester::requestUrl, worker,
+                     &NetworkTask::run);
 
-        auto *reply = [&]() -> QNetworkReply * {
-            switch (data->requestType_)
-            {
-                case NetworkRequestType::Get:
-                    return NetworkManager::accessManager.get(data->request_);
+    emit requester.requestUrl();
+}
 
-                case NetworkRequestType::Put:
-                    return NetworkManager::accessManager.put(data->request_,
-                                                             data->payload_);
+// First tried to load cached, then uncached.
+void loadCached(std::shared_ptr<NetworkData> &&data)
+{
+    QFile cachedFile(getPaths()->cacheDirectory() + "/" + data->getHash());
 
-                case NetworkRequestType::Delete:
-                    return NetworkManager::accessManager.deleteResource(
-                        data->request_);
+    if (!cachedFile.exists() || !cachedFile.open(QIODevice::ReadOnly))
+    {
+        // File didn't exist OR File could not be opened
```

Comment:
This comment repeats the code above it. I don't think it helps, removing it is probably best.
```suggestion
```